build.gradle

```diff
@@ -519,12 +520,32 @@ forbiddenApisTest.enabled = false
  * Maven Publish *
  *****************/
 
+void addManifest(Jar jar) {
+    jar.manifest {
+        attributes 'Implementation-Title': project.name
+        attributes 'Implementation-Vendor': 'HiveMQ'
+        attributes 'Implementation-Version': project.version
+    }
+}
+
+task embeddedJar(type: Jar) {
+    group 'build'
+
+    from sourceSets.main.output
+    addManifest(it)
+    exclude('META-INF/INDEX.LIST', 'META-INF/*.SF', 'META-INF/*.DSA', 'META-INF/*.RSA')
+}
+
+task sourceJar(type: Jar) {
```

Comment:
convention for this name is sourcesJar
We can use the following when rebasing on the latest changes which update the gradle version
```
java {
    withJavadocJar()
    withSourcesJar()
}
```

Reply:
nice 👍